### ISSUE TITLE
Troubleshooting Intel AMT  CIRA Configuration/Activation in MeshCentral Public Server Setup

### ISSUE THREAD

**Issue by amoljagdalepucsd** [Ylianst/MeshCentral#6049]
Hi Community ,

I am running a MeshCentral public server on the internet and trying to activate Intel AMT. So far, I have successfully activated Intel AMT on a remote system with a TLS certificate. However, based on my understanding, to enable AMT on a public server, we need to enable/activate CIRA. For that purpose, I've set the policy to fully automatic at the group level, so it should automatically configure CIRA and establish a connection to the MeshCentral Intel AMT server. However, CIRA is not activating, and Intel AMT is not connected. The 'Intel AMT connected' status should be displayed under the general tab once AMT is connected, and the connect button under the Intel AMT tab should also become enabled. But nothing is happening. Can anyone please help me with this?

Additionally, I'm wondering if I missed any configurations that could be causing the issue. Could someone please advise on what configurations might be necessary? Furthermore, I'd appreciate any guidance on how to debug this issue effectively. Thank you.

<img width="283" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/48832403/a31128b7-9219-4e38-a295-45f989f10a74">
<img width="595" alt="image" src="https://github.com/Ylianst/MeshCentral/assets/48832403/6de053c4-7a44-4d15-8650-d3498076dbb2">

**Comment by si458** [comment]
**YOU HAVENT FOLLOWED THE BUG TEMPLATE REPORT WHICH HAS INFORMATION WE NEED TO HELP YOU**
what nodejs version are you using?
what meshcentral version are you using?
are you using a reverse proxy (like traefik/nginx)?
what intel AMT version is the remote device?

what is the outputs from running the following commands from the `console` tab of the device?
`amt`
`amtevents`
`amtconfig` - this will config amt again, give it a few mins to run and display output

**Comment by amoljagdalepucsd** [comment]
Nodejs : v20.12.1 ,
MeshCentral : 1.0.85 ,
reverse proxy : We have nginx to before MeshCentral server ,
amt version : 11.0.18 ,

>amt
{
  core-ver: 1
  OsHostname: "LAP-084"
  Flags: 4
  MeiVersion: "100.0.0.5124"
  Versions: {
    Flash: "11.0.18"
    Netstack: "11.0.18"
    AMTApps: "11.0.18"
    AMT: "11.0.18"
    Sku: "16392"
    VendorID: "8086"
    Build Number: "1002"
    Recovery Version: "11.0.18"
    Recovery Build Num: "1002"
    Legacy Mode: "False"
  }
  UUID: "MY UID"
  ProvisioningMode: 1
  ProvisioningState: 2
  net0: {
    enabled: 1
    dhcpEnabled: 1
    dhcpMode: "PASSIVE"
    mac: "D4:81:D7:A4:94:5A"
    address: "0.0.0.0"
  }
  net1: {
    enabled: 1
    dhcpEnabled: 1
    dhcpMode: "PASSIVE"
    mac: "28:16:AD:8A:1E:1B"
    address: "0.0.0.0"
  }
  DnsSuffix: "abc.com"
}

>amtevents 
04:13:00, LMS tunnel start.
04:13:03, Checking Intel AMT state...
04:13:12, Intel AMT connected.
04:13:42, Removing CIRA periodic trigger.
04:13:42, Removing MPS server.
04:13:45, Created new MPS server.
04:13:46, Created new MPS policy.
04:13:55, Done.
04:13:55, LMS tunnel closed.

>amtconfig 
05:07:02, LMS tunnel start.
handleApfJsonControl 1.0 :{"action":"console","msg":"Checking Intel AMT state..."}
05:07:04, Checking Intel AMT state...
handleApfJsonControl 1.0 :{"action":"console","msg":"Intel AMT connected."}
05:07:13, Intel AMT connected.

**Comment by si458** [comment]
you are still using an old version of meshcentral, you need to update, 1.1.22 is the latest which has AMT fixes in place

**Comment by amoljagdalepucsd** [comment]
I am planning to update it, but as it is hosted on a server with some changes, it will take time to update. CIRA should work with this version as well, I guess, because I saw @Ylianst s videos, and he was able to activate CIRA with an older version.

**Comment by si458** [comment]
yes but the was amt issues using node 16 or above, which was only recently fixed in like 1.1.20 if i remember?
so you need to update first, then report back if your still having issues

**Comment by si458** [comment]
that commit is just 1 fix out of about 50
you need to update to 1.1.22 and then re-test and report back,
applying patches MIGHT not work

**Comment by amoljagdalepucsd** [comment]
@si458 , Thanks for suggestions. I will update and report back.
I am now able to activate CIRA, and Intel AMT is also connected to my development server and able to see all info in Intel AMT tab. 
However, the remote desktop session is not connecting under the Intel AMT tab. Additionally, I noticed that CIRA gets disconnected after some time.
<img width="953" alt="AMT TAB" src="https://github.com/Ylianst/MeshCentral/assets/48832403/6f541a3a-8352-4279-86aa-1918fcd4ffbc">
<img width="526" alt="AMT connect" src="https://github.com/Ylianst/MeshCentral/assets/48832403/f95dbcd2-9058-48e0-b822-a7ce6770393d">

**Comment by si458** [comment]
@amoljagdalepucsd, so ur one step closer, I'm guessing u hand patched rather than updating?
again, update to the latest version and then try again.
Or restart meshcentral and power cycle the remote device (unplug its power, wait 5 secs, plug it bk in)

**Comment by amoljagdalepucsd** [comment]
restart server and power cycle the device
getting below error in browser console.

'wss://anmoldev.com/webrelay.ashx?p=2&host=node//xVPkRQqgfwamD4ogFKbEQZvFlw0uz$EGxnctJJ9fXi167@unXVy5hTygO$DJBsAh&port=16994&tls=0&serverauth=1' failed: WebSocket is closed before the connection is established.
